test/notfoundigest/handler-external.test.js

```diff
@@ -0,0 +1,185 @@
+  };
+
+  beforeEach('setup', () => {
+    context = {
+      log: console,
+      dataAccess: mockDataAccess,
+      env: {
+        RUM_DOMAIN_KEY: 'uber-key',
+      },
+    };
+  });
+
+  afterEach('clean', () => {
+    sandbox.restore();
+  });
+
+  it('builds and sends the slack message when there is an org config and a 404 audit stored for the site', async () => {
+    context.rumApiClient = {
+      create404Backlink: sandbox.stub().resolves(backlink),
+    };
+    const channel = 'channel1';
+    context.slackClients = {
+      ADOBE_EXTERNAL: { postMessage: sandbox.stub().resolves({ channelId: channel, threadId: 'ts-1' }) },
+    };
+    const resp = await notFoundExternalDigestHandler({}, context);
+    expect(resp.status).to.equal(204);
+  });
+
+  it('builds no message when there is no audit', async () => {
+    const noAuditContext = { ...context };
+    noAuditContext.dataAccess = { ...context.dataAccess };
+    noAuditContext.dataAccess.getSitesByOrganizationIDWithLatestAudits = () => [];
+    const resp = await notFoundExternalDigestHandler({}, noAuditContext);
+    expect(resp.status).to.equal(204);
+  });
+
+  it('returns 500 if the initial slack api fails', async () => {
+    context.slackClients = {
+      ADOBE_EXTERNAL: { postMessage: sandbox.stub().rejects(new Error('error')) },
+    };
+    const resp = await notFoundExternalDigestHandler({}, context);
+    expect(resp.status).to.equal(500);
+  });
+
+  it('continues if just one slack api call failed', async () => {
+    const channel = 'channel1';
+    context.rumApiClient = {
+      create404Backlink: sandbox.stub().resolves(backlink),
+    };
+    context.slackClients = {
+      ADOBE_EXTERNAL: { postMessage: sandbox.stub().onFirstCall().resolves({ channel, ts: 'ts-1' }) },
+    };
+    context.slackClients.ADOBE_EXTERNAL.postMessage.onSecondCall().rejects(new Error('error'));
+
+    const resp = await notFoundExternalDigestHandler({}, context);
+    expect(resp.status).to.equal(204);
+  });
+
+  it('builds and sends the slack message when there is an site config and a 404 audit stored for the site', async () => {
+    const channel = 'channel1';
+    context.rumApiClient = {
+      create404Backlink: sandbox.stub().resolves(backlink),
+    };
+    context.slackClients = {
+      ADOBE_EXTERNAL: { postMessage: sandbox.stub().resolves({ channel, ts: 'ts-1' }) },
+    };
+    const siteContext = { ...context };
+    const newOrgData = {
+      ...organizationData,
+      getConfig: () => ({
+        alerts: [{
+          type: '404',
+          byOrg: false,
+        }],
+      }),
+    };
+    siteContext.dataAccess.getOrganizations = sinon.stub().resolves([newOrgData]);
+    const resp = await notFoundExternalDigestHandler({}, siteContext);
+    expect(resp.status).to.equal(204);
+  });
+
+  it('builds and sends the slack message when there is an site config and a 404 audit stored for the site and no backlink', async () => {
```

Comment:
we need at least one E2E test scenario where we provide real value from fixtures like: https://github.com/adobe/spacecat-audit-post-processor/blob/main/test/fixtures/audit-results.js

the fixture should at least contain 3 orgs (with sites and audits): one is with `byOrg=true`, one is with `byOrg=false`, and one with not alert config for 404. The test should check if the `slackClient` is called properly with expected messages